### PR TITLE
Add badge concept draft docs

### PR DESCRIPTION
**Pull request by nidico**
We decided to put the current state of thoughts into the docs, even if
modelling isn't clear yet, as that will be postponed by a couple of
weeks.

I decided to add a drafts section in docs for such things.

### CODE REVIEW

File: docs/drafts/badges.rst [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+The various badges, categories and tags can be categorized as the
+following:
+
+
+Allowed targets
+    What can be badged?
+
+
+Locality
```
Comment (xi) [inline docs/drafts/badges.rst]: I would rename this to "scope".

**Comment by xi** [comment]
waiting for further feedback

File: docs/drafts/badges.rst [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+to streamline these entities into more consistent or possibly independent
+concepts in Adhocracy 3. Some aspects shall not be implemented in A3.
+
+This document can be replaced once the concepts are ready in A3.
+
+
+Badge features in Adhocracy 2
+-----------------------------
+
+The various badges, categories and tags can be categorized as the
```
Comment (MagdaN) [inline docs/drafts/badges.rst]: What is the difference between badges, categories and tags?

File: docs/drafts/badges.rst [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    What can be badged?
+
+
+Locality
+    Do badges exist globally or locally only?
+
+Exclusivity
+    Can only one badge out of a certain group be assigned to a target
+    object?
+
```
Comment (MagdaN) [inline docs/drafts/badges.rst]: Is that the same as: "Can a target object only have one badge?"

File: docs/drafts/badges.rst [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+Icon
+    If shown as thumbnail, which icon should be shown?
+
+Visibility
+    Should it be shown at all in listings?
+
+
+Impact
+    Effect on mixed list sortings
+
```
Comment (MagdaN) [inline docs/drafts/badges.rst]: What is meant by that?

File: docs/drafts/badges.rst [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+Implicit user role
+    If a user has this badged, which additional role should she have?
+    (we should drop this)
+
```
Comment (MagdaN) [inline docs/drafts/badges.rst]: What is meant by that?

File: docs/drafts/badges.rst [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    Assign a certain behaviour to a badge (research project at HHU)
+
+
+Badges, categories and tags in A2
+---------------------------------
+
+Some remarkable aspects of badge-like entities in A2:
+
+
+Default
```
Comment (MagdaN) [inline docs/drafts/badges.rst]: What dou you mean by default? Is that the default behaviour of badges, categories and tags? Or is this some abstrct class, that all badge-like entities in A2 inherit?

File: docs/drafts/badges.rst [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+
+Requirements for badges in A3
+-----------------------------
+
+(this is incomplete)
+
+-   Allow to define which badges can be assigned to a certain
+    resource with which rules (see features above).
+
```
Comment (MagdaN) [inline docs/drafts/badges.rst]: Which feature from above do you mean?

File: docs/drafts/badges.rst [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+-   It must be possible to freely define multiple available badge groups.
+
```
Comment (MagdaN) [inline docs/drafts/badges.rst]: What do you mean by badge group?

File: docs/drafts/badges.rst [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+-   Allow to define badges globally and locally. It should be possible
+    to restrict and extend the available badges locally.
+
+-   NTH: Badges can be connected through a common taxonoy, i.e. if a
+    local process wants to call a badge `Umweltpolitik` it can be
+    connected to a global badge `Umwelt`.
+
```
Comment (MagdaN) [inline docs/drafts/badges.rst]: Isn't that like structuring the badges hierarchically?

File: docs/drafts/badges.rst [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+-   All badges shall be indexed and can be used in pool queries.
+
+
+Example
++++++++
+
+Some proposal resource might be badged as the following:
+
+-   Badge group `decision_state`:
+
```
Comment (MagdaN) [inline docs/drafts/badges.rst]: So this is something like a value from a predefined list of categories, right?

File: docs/drafts/badges.rst [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+        - available badges "beschlossen", "abgelehnt"
+        - exclusive
+        - creatable by admins (not really necessary, because hardcoded)
+        - assignable by moderators
+
+-   Badge group `topic`:
+
```
Comment (MagdaN) [inline docs/drafts/badges.rst]: This is something like creating a list of tags that users can choose from when tagging their proposals?

**Comment by MagdaN** [comment]
So to sum up: you don't want to implement the different concepts 'tags', 'categories', 'badges' seperately but instead only implement just one concept and assign to each instance of a badge whether it exclusive, hierarchical, has a pages (wouldn't that page be a list of badget items filteres by the badge? ), has a icon, has a image, Can be created and assigned by specific useres, can exist globally and additonally per instance, and so on?

File: docs/drafts/badges.rst (hunk 2 above)
Comment (nidico) [inline docs/drafts/badges.rst]: This is defined briefly in the section *Badges, categories and tags in A2* below.

File: docs/drafts/badges.rst (hunk 1 above)
Comment (nidico) [inline docs/drafts/badges.rst]: Agreed.

File: docs/drafts/badges.rst (hunk 3 above)
Comment (nidico) [inline docs/drafts/badges.rst]: Basically yes, but per "badge group". If there's a badge group "decided state" and a badge group "implemented state" (both exclusive), then the referred object can have up to one badge out of the possible "decided state" badges and up to one badge out of the possible "implemented state".

File: docs/drafts/badges.rst [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+-   Allow to define which badges can be assigned to a certain
+    resource with which rules (see features above).
+
+-   It must be possible to freely define multiple available badge groups.
+
+-   Allow to define badges globally and locally. It should be possible
+    to restrict and extend the available badges locally.
+
+-   NTH: Badges can be connected through a common taxonoy, i.e. if a
```
Comment (nidico) [inline docs/drafts/badges.rst]: Typo: *taxonomy*

File: docs/drafts/badges.rst (hunk 4 above)
Comment (nidico) [inline docs/drafts/badges.rst]: see [A2 moderator docs](https://adhocracy.de/static/help/anleitung-fuer-moderatorinnen#1-4-1-kategorien-als-thematische-sortierung-in--vorschl-gen-) (sorry, too lazy atm)

File: docs/drafts/badges.rst (hunk 5 above)
Comment (nidico) [inline docs/drafts/badges.rst]: You can attach roles to user badges in A2. If a user is badged with such a badge, she also has all permissions of the connected role.

File: docs/drafts/badges.rst (hunk 6 above)
Comment (nidico) [inline docs/drafts/badges.rst]: I originally had a matrix of "A2 badge types" vs "Features". In order to summarize that, I artificially declared a "common" behavior of most badge types / categories / tags and explicitly highlighted the distinctiveness for certain entities below.

File: docs/drafts/badges.rst (hunk 7 above)
Comment (nidico) [inline docs/drafts/badges.rst]: All.

File: docs/drafts/badges.rst (hunk 8 above)
Comment (nidico) [inline docs/drafts/badges.rst]: Different to A2 where the existing badge types are fixed, we want to freely define and configure multiple possible badge groups, out of which badges can be assigned to the resource.

In the UI, each badge group typically has a header in assignment forms / facet selections.

File: docs/drafts/badges.rst (hunk 9 above)
Comment (nidico) [inline docs/drafts/badges.rst]: Maybe. The concrete implementation is unclear as of now.

File: docs/drafts/badges.rst (hunk 11 above)
Comment (nidico) [inline docs/drafts/badges.rst]: Yes.

File: docs/drafts/badges.rst (hunk 10 above)
Comment (nidico) [inline docs/drafts/badges.rst]: Yes.

**Comment by nidico** [comment]
> So to sum up: you don't want to implement the different concepts 'tags', 'categories', 'badges' seperately but instead only implement just one concept and assign to each instance of a badge whether it exclusive, hierarchical, has a pages, has a icon, has a image, Can be created and assigned by specific useres, can exist globally and additonally per instance, and so on?

Yes. Or at least the relevant features out of that list.

> has pages (wouldn't that page be a list of badged items filtered by the badge?)

Yes, mostly, but it may also have an own image or description text (at least category pages do in A2 - see [example](https://urbanliving.berlin.de/i/urbanliving/category).

**Pull request merged**